### ISSUE TITLE
实验 MFTCoder 的效果总是不尽人意

### ISSUE THREAD

**Issue by Chaochao2020** [codefuse-ai/MFTCoder#64]
你好，我使用了三个开源问答数据集，分别为 Squad、Commonsense、ai2_arc 来进行多任务微调，但我迄今为止微调三次试验结果都不如单次微调甚至不如混合微调，我使用 mistral_v0.2 模型对三个数据集分别进行微调、混合微调、不微调、MFTcoder 微调进行比较，下面是我的实验结果：
<img width="582" alt="image" src="https://github.com/codefuse-ai/MFTCoder/assets/59333967/b5242c0f-0ed3-4063-917c-d109d97ba9e1">

所以我在怀疑是我微调参数是否出了问题，或者说这三个问答子任务相关性并不强所以导致这种现象。所以在此提出 issue 来向你们请教。

下面是我的第三次 mftcoder 微调配置：
```bash
{
    "data_paths": "[data/ai2_arc,data/squad,data/commonsense_qa]",
    "output_dir": "model/mistral/single_train_2024-06-13/lora_adaptor",
    "tb_dir": "model/mistral/single_train_2024-06-13/tensorboard",
    "pretrained_model_path": "AI-ModelScope/Mistral-7B-Instruct-v0___2",
    "model_type": "mistral",
    "load_raw_dataset": true,
    "data_split": "98,2,0",
    "padding_mode": "padding",
    "use_dynamic_padding": true,
    "tokenize_mode": "sft",
    "tokenizer_type": "AutoTokenizer",
    "weighted_loss_mode": "case3",
    "attn_implementation": "flash_attention_2",
    "seq_length": 1024,
    "seed": 1234,
    "peft_type": "lora",
    "quantization": null,
    "lora_rank": 64,
    "lora_alpha": 128,
    "lora_dropout": 0.05,
    "per_device_train_batch_size": 16,
    "per_device_eval_batch_size": 32,
    "learning_rate": 5e-5,
    "min_lr": 5e-6,
    "weight_decay": 0.1,
    "gradient_accumulation_steps": 2,
    "lr_scheduler_type": "cosine",
    "num_warmup_steps": 300,
    "num_train_epochs": 2,
    "log_interval": 10,
    "checkpointing_steps": 100,
    "evaluation_steps": 100,
    "max_train_steps": null,
    "epoch_checkpointing": true,
    "shuffle_before_split": true,
    "early_stopping": true,
    "early_stopping_stall_num": 5,
    "saving_limit": null,
    "target_modules":["q_proj", "k_proj", "v_proj", "o_proj", "gate_proj", "down_proj", "up_proj"]
}

此外你的配置选项中有一些参数并没有说明，例如 data weight 等，我觉得对于框架使用者来说还是完善一些更好，

**Comment by lxFang2022** [comment]
> 你好，我使用了三个开源问答数据集，分别为 Squad、Commonsense、ai2_arc 来进行多任务微调，但我迄今为止微调三次试验结果都不如单次微调甚至不如混合微调，我使用 mistral_v0.2 模型对三个数据集分别进行微调、混合微调、不微调、MFTcoder 微调进行比较，下面是我的实验结果： <img alt="image" width="582" src="https://private-user-images.githubusercontent.com/59333967/339256726-b5242c0f-0ed3-4063-917c-d109d97ba9e1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE0NTg1NTIsIm5iZiI6MTcyMTQ1ODI1MiwicGF0aCI6Ii81OTMzMzk2Ny8zMzkyNTY3MjYtYjUyNDJjMGYtMGVkMy00MDYzLTkxN2MtZDEwOWQ5N2JhOWUxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzIwVDA2NTA1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThlZWZiYzRjNDVhZGZhZGNkYWFhODE1Nzk0NDlkNmU5MjgxMWY3OGUyMWUwMDU0MzdjOWViODhiMGNhMTc0YmQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.5uEQUFgUhgtEVAgvkhUiyVfhpf65Fi3JnHFsxNGSPik">
> 
> 所以我在怀疑是我微调参数是否出了问题，或者说这三个问答子任务相关性并不强所以导致这种现象。所以在此提出 issue 来向你们请教。 此外你的配置选项中有一些参数并没有说明，例如 data weight 等，我觉得对于框架使用者来说还是完善一些更好。
> 
> 下面是我的第三次 mftcoder 微调配置：
> 
> ```shell
> {
>     "data_paths": "[data/ai2_arc,data/squad,data/commonsense_qa]",
>     "output_dir": "model/mistral/single_train_2024-06-13/lora_adaptor",
>     "tb_dir": "model/mistral/single_train_2024-06-13/tensorboard",
>     "pretrained_model_path": "AI-ModelScope/Mistral-7B-Instruct-v0___2",
>     "model_type": "mistral",
>     "load_raw_dataset": true,
>     "data_split": "98,2,0",
>     "padding_mode": "padding",
>     "use_dynamic_padding": true,
>     "tokenize_mode": "sft",
>     "tokenizer_type": "AutoTokenizer",
>     "weighted_loss_mode": "case3",
>     "attn_implementation": "flash_attention_2",
>     "seq_length": 1024,
>     "seed": 1234,
>     "peft_type": "lora",
>     "quantization": null,
>     "lora_rank": 64,
>     "lora_alpha": 128,
>     "lora_dropout": 0.05,
>     "per_device_train_batch_size": 16,
>     "per_device_eval_batch_size": 32,
>     "learning_rate": 5e-5,
>     "min_lr": 5e-6,
>     "weight_decay": 0.1,
>     "gradient_accumulation_steps": 2,
>     "lr_scheduler_type": "cosine",
>     "num_warmup_steps": 300,
>     "num_train_epochs": 2,
>     "log_interval": 10,
>     "checkpointing_steps": 100,
>     "evaluation_steps": 100,
>     "max_train_steps": null,
>     "epoch_checkpointing": true,
>     "shuffle_before_split": true,
>     "early_stopping": true,
>     "early_stopping_stall_num": 5,
>     "saving_limit": null,
>     "target_modules":["q_proj", "k_proj", "v_proj", "o_proj", "gate_proj", "down_proj", "up_proj"]
> }
> ```

您好 我也想用mistral_v0.2 模型进行微调，但是遇到了一点问题，可以加个联系方式交流一下吗？我的邮箱是791785837@qq.com

**Comment by XiaoMaGe-hero** [comment]
我使用自己场景的数据集（每个任务数据量 1k ～ 20k）微调，四个任务的收敛速度、收敛效果也是更差，或许需要使用更多epoch 训练？
